packages/@okta/vuepress-site/docs/guides/pwd-optional-widget-sign-in-email/main/aspnet/integrationsteps.md

```diff
@@ -0,0 +1,58 @@
+The following code defines an endpoint that retrieves the current Sign-In Widget configuration details from session state and appends the `OTP` and `state` values from the callback to it. If it cannot find the configuration details in session state, it assumes the callback has not come from the same browser and prompts the user accordingly.
+
+```csharp
+public async Task<ActionResult> Callback(
+   string state, string otp, string error = null, string error_description = null)
+{
+   // handle errors if error is not null
+
+   SignInWidgetConfiguration siwConfig = Session["siwConfig"] as SignInWidgetConfiguration;
+   if (siwConfig != null)
+   {
+       siwConfig.Otp = otp;
+       siwConfig.State = state;
+       return View("SignInWidget", siwConfig);
+   }
+
+   return View(new MagicLinkCallbackModel {
+      Message = $"Please enter the OTP '{otp}' in the original browser tab to finish the flow."
+   });
+}
+```
+
+> **Note**: For more information on magic links and OTP, including customizations and complete user journeys, see [Email Magic Links Overview](/docs/guides/email-magic-links-overview/main/).
+
+### 4. Your app handles an authentication success response
+
+After the user successfully verifies their identity, Identity Engine sends an interaction code in a query parameter to `${signInRedirectURI}`. For example, `http://localhost:44314/interactioncode/callback?interaction_code=2JFmObNY8snovJP6_UK5gI_l7RQ-....`
+
+Create an endpoint that calls `idxClient.RedeemInteractionCodeAsync()` to exchange the interaction code for access tokens and `AuthenticationHelper.GetIdentityFromTokenResponseAsync()` to retrieve the OIDC claims information about the user and pass them into your application. The user has now signed in.
```

Comment:
"retrieve the OIDC claims information about the user" is a little awkward. How about " retrieve the user's OIDC claims information" ?

Reply:
"pass them into your application" -> by them, are you talking about the OIDC claims information, or the access tokens? If the former, you should change "them" to "it". If the latter, you should make that clearer.

Reply:
Fixed. @robertnicolo-okta  Are you happy for me to make this word change in all our pwd-optional guides? We have the same text everywhere.

Reply:
Yes, @danielmaharry-okta